### PR TITLE
#83099 replace HttpClientExtensions with injected service

### PR DESCRIPTION
**Pull request by tabman83**

### CODE REVIEW

File: src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs [hunk 1]
```diff
@@ -21,14 +21,15 @@ public class OidcAuthenticationHandler : AuthenticationHandler, IAuthenticationH
         protected OidcAuthenticationToken OidcAuthenticationToken = new OidcAuthenticationToken();
         protected readonly OidcAuthenticationConfig OidcAuthenticationConfig;
         private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
-        protected readonly IHttpClientFactory HttpClientFactory;
+        protected readonly IHttpSender HttpSender
+            ;
```
Comment (dawidkeshop) [inline src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs:25]: This semicolon looks very alone. :(

File: src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs [hunk 2]
```diff
@@ -21,14 +21,15 @@ public class OidcAuthenticationHandler : AuthenticationHandler, IAuthenticationH
         protected OidcAuthenticationToken OidcAuthenticationToken = new OidcAuthenticationToken();
         protected readonly OidcAuthenticationConfig OidcAuthenticationConfig;
         private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
-        protected readonly IHttpClientFactory HttpClientFactory;
+        protected readonly IHttpSender HttpSender
```
Comment (dawidkeshop) [inline src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs:24]: Aren't we using `_httpSender` convention?

File: src/CaptainHook.EventHandlerActor/Handlers/HttpSender.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            });
+
+        }
+
+        /// <summary>
+        /// Executes the supplied func with reties and reports on it if something goes wrong ideally to BigBrother
+        /// </summary>
+        /// <param name="makeTheCall"></param>
+        /// <returns></returns>
+        private static async Task<HttpResponseMessage> RetryRequest(
```
Comment (dawidkeshop) [inline src/CaptainHook.EventHandlerActor/Handlers/HttpSender.cs:78]: Is this method used anywhere?

File: src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs (hunk 2 above)
Comment (tabman83) [reply src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs:24]: Just had it capital like other protected members there

Comment (mszymeshop) [reply src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs:24]: for private fields

File: src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs (hunk 1 above)
Comment (tabman83) [reply src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs:25]: :D

File: src/CaptainHook.EventHandlerActor/Handlers/HttpSender.cs (hunk 3 above)
Comment (tabman83) [reply src/CaptainHook.EventHandlerActor/Handlers/HttpSender.cs:78]: Line 48 same file

Comment (mszymeshop) [reply src/CaptainHook.EventHandlerActor/Handlers/HttpSender.cs:78]: in line `48`

**Review by mszymeshop — APPROVED** [review]

File: src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs (hunk 2 above)
Comment (dawidkeshop) [reply src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs:24]: I thought we're using it for all fields, at least I did that way. We definitely need a team-level set of rules to be configured.

**Review by dawidkeshop — APPROVED** [review]

File: src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs (hunk 1 above)
Comment (tabman83) [reply src/CaptainHook.EventHandlerActor/Handlers/Authentication/OidcAuthenticationHandler.cs:25]: I am not even able to type an emoticon, let alone getting a semicolon right!

**Pull request merged**